From 35502edb91715a0935781587a12fe6e3bafed578 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 09 Dec 2019 09:18:37 +0200
Subject: [PATCH] [SW-7263] ppboot: dump reset source

It is useful if one wants to know cause of reset

Note: image binary size increased by 200 Bytes

Change-Id: Ia0e3550131c88c5d46f0ff9205e77d4006fa3643
---

diff --git a/app/ppboot.c b/app/ppboot.c
index b65bfb9..50f5d3e 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -53,6 +53,36 @@
 	spi_copy((void *)PREBOOT_TEXT_BASE, preboot_offset, PREBOOT_SIZE, name);
 }
 
+static void rst_src_dump(void)
+{
+	unsigned int rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
+	char *cause;
+
+	switch (rst_src) {
+	case HL_COLD_RST:
+		cause = "cold";
+		break;
+	case HL_MANUAL_RST:
+		cause = "manual";
+		break;
+	case HL_PRSTN_RST:
+		cause = "prstn";
+		break;
+	case HL_WD_RST:
+		cause = "watch dog";
+		break;
+	case HL_SW_ALL_RST:
+		cause = "software all";
+		break;
+	default:
+		cause = "unknown";
+		break;
+	}
+	preboot_puts_raw("Last reset cause is: ");
+	preboot_puts_raw(cause);
+	preboot_puts(" reset\n");
+}
+
 static bool hl_pcie_link_up(void)
 {
 	unsigned int prstn, smlh, rdlh, rst_src;
@@ -108,6 +138,7 @@
 	preboot_sync_btl();
 	hl_uart_init();
 	ppboot_pcie_init();
+	rst_src_dump();
 
 	/* check if we are done here */
 	if (hl_pcie_init_only())
